SAP/cloud-commerce-spartacus-storefront
# #184 second part of routing implementation: pipes

This PR includes:
- Pipes generating navigation links (`cxPath` and `cxDynamicPath`)
- All dynamic urls from CMS are translated using `cxDynamicPath` pipe (i.e. whole top CMS-driven navigation)
How to test:
  1. please set in app.module config:
```
      routesConfig: {
        translations: {
          en: {
            storeFinder: ['store-locator'],
            myAccount_orders: ['test/order-history'],
            category: [
              'category/:categoryCode',
              'brands/:brandCode/:brandName'
            ]
          }
        }
      }
```
  2. as **logged in** user and **having dev-com17 backend** see top-navigation links:
    - Find a store
    - My Account (dropdown) -> Order history
    - Brands (category dropdown) -> Canon

- Link to product page is translated using `cxPath` pipe (only in `product-list-item.component`, as representative example)
How to test: 
  1. please set in app.module config:
```
      routesConfig: {
        translations: {
          en: {
            product: ['p/:productCode/:name']
          }
        }
      }
```
or use also **parameter names mapping**:
```
      routesConfig: {
        translations: {
          en: {
            product: ['p/:productCode/:productName']
          }
        },
        parameterNamesMapping: {
          product: {
            productName: 'name'
          }
        }
      }
```
  2. go to category page (product list view) and see link on image and product title


For clarity purpose, all other usages of configurable routes and cxPath pipes will be included in next PR (50+ files).
